statistics/handle/lru_cache.go

```diff
@@ -0,0 +1,368 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"container/list"
+	"sync"
+
+	"github.com/pingcap/tidb/statistics"
+)
+
+type statsInnerCache struct {
+	sync.RWMutex
+	elements map[int64]*lruMapElement
+	// lru maintains item lru cache
+	lru *innerItemLruCache
+}
+
+func newStatsLruCache(c int64) *statsInnerCache {
+	s := &statsInnerCache{
+		elements: make(map[int64]*lruMapElement),
+		lru:      newInnerLruCache(c),
+	}
+	s.lru.onEvict = s.onEvict
+	return s
+}
+
+type innerItemLruCache struct {
+	capacity     int64
+	trackingCost int64
+	elements     map[int64]map[int64]*list.Element
+	cache        *list.List
+	onEvict      func(tblID int64)
+}
+
+func newInnerLruCache(c int64) *innerItemLruCache {
+	return &innerItemLruCache{
+		capacity: c,
+		cache:    list.New(),
+		elements: make(map[int64]map[int64]*list.Element, 0),
+	}
+}
+
+type lruCacheItem struct {
+	tblID         int64
+	id            int64
```

Comment:
Do we just use `Index.ID` and `Column.ID` to set `lruCacheItem.id` here?

Reply:
We will differentiate between index and column in the next pr.

Reply:
right